src/main/java/rearth/oritech/block/entity/machines/interaction/LaserArmBlockEntity.java

```diff
@@ -95,112 +129,107 @@ public void markDirty() {
     private BaseAddonData addonData = MachineAddonController.DEFAULT_ADDON_DATA;
     public int areaSize = 1;
     public int yieldAddons = 0;
+    public int hunterAddons = 0;
+    public boolean hasCropFilterAddon = false;
     
     // config
     private final int range = Oritech.CONFIG.laserArmConfig.range();
+
+    private Vec3d laserHead;
     
     // working data
     private BlockPos targetDirection;
     private BlockPos currentTarget;
+    public HunterTargetMode hunterTargetMode = HunterTargetMode.HOSTILE_ONLY;
+    private LivingEntity currentLivingTarget;
     private long lastFiredAt;
     private int progress;
     private int targetBlockEnergyNeeded = BLOCK_BREAK_ENERGY;
     private boolean networkDirty;
     private boolean redstonePowered;
     private ArrayDeque<BlockPos> pendingArea;
+    private ArrayDeque<LivingEntity> pendingLivingTargets;
     
     // needed only on client
     public Vec3d lastRenderPosition;
     private PlayerEntity laserPlayerEntity = null;
     
     public LaserArmBlockEntity(BlockPos pos, BlockState state) {
         super(BlockEntitiesContent.LASER_ARM_ENTITY, pos, state);
+        laserHead = Vec3d.of(pos.up()).add(0.5, 0.55, 0.5);
     }
     
     @Override
     public void tick(World world, BlockPos pos, BlockState state, LaserArmBlockEntity blockEntity) {
-        if (world.isClient() || !isActive(state) || redstonePowered || currentTarget == null || energyStorage.getAmount() < energyRequiredToFire())
+        if (world.isClient() || !isActive(state) || redstonePowered || (hunterAddons == 0 && currentTarget == null) || energyStorage.getAmount() < energyRequiredToFire())
```

Comment:
I've noticed that the laser arm UI doesnt update the current energy amount when it's not active. E.g. when it's placed and charging up. This is probably caused by this section here

Reply:
I reorganized this so that it will always call updateNetwork() if the network is marked as dirty as long as it's server side and the laser arm is assembled.

Reply:
I did notice that the "isActive" check doesn't seem to be working, though. The laser arm is getting charged even before it's assembled. I also checked the assembler, and it was also getting charged before it was assembled.